### ISSUE TITLE
alprd crashing

### ISSUE THREAD

**Issue by edwardscdr** [openalpr/openalpr#225]
When running alprd daemon it will crash regularly (every hour or so) with error:
```*** Error in `alprd': double free or corruption (!prev): 0x71584f18 ***```
I am running on a Raspberry Pi 2 (Jessie) with opencv 3, tesseract 3.04, leptonica 1.72. Before crashing it is working correctly.
Thanks

**Comment by edwardscdr** [comment]
I'm using latest openalpr, cloned today

**Comment by matthill** [comment]
Did you figure out what the problem was?

**Comment by edwardscdr** [comment]
Yes thanks, not a problem with openalpr. It was a bad SD card in the Raspberry Pi causing crashing. Working fine with another one.

**Comment by edwardscdr** [comment]
No spoke to soon. The error returns after a few hours running - *** Error in `alprd': double free or corruption (!prev): 0x71588120 ***

**Comment by matthill** [comment]
I have a RPI2 as well, I can test on it at some point.  I suspect it's probably a recent bug caused by differences between 64-bit and 32-bit or ARM.

If you want to try debugging on your own, the process I usually follow is:

Turn on debugging by adding:

set(CMAKE_BUILD_TYPE Debug)

to the top of the CMakeLists.txt file in the root directory and then recompiling

Then run alprd in gdb

gdb ./alprd
(gdb) set args -f
(gdb) run

Eventually if the program crashes, you'll be able to print out a stack trace (backtrace full) and other variable information at the time/location of the crash.

**Comment by edwardscdr** [comment]
Hi Matt,

I did that and got an error straight away:

gdb ./alprd
GNU gdb (Raspbian 7.7.1+dfsg-5) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "arm-linux-gnueabihf".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./alprd...(no debugging symbols found)...done.
(gdb) set args -f
(gdb) run
Starting program: /home/pi/builds/openalpr/src/alprd -f
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".
Cannot access memory at address 0x0

Program received signal SIGILL, Illegal instruction.
0x74249d08 in ?? () from /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0
(gdb)

**Comment by edwardscdr** [comment]
Also get error sometimes:

Error: signal 11:
alprd(_Z16segfault_handleri+0x20)[0x3ef94]
/lib/arm-linux-gnueabihf/libc.so.6(__default_sa_restorer_v2+0x0)[0x74409160]

**Comment by matthill** [comment]
I think the error with GDB is because of this:

    Reading symbols from ./alprd...(no debugging symbols found)...done

.  You need to add this to the top of CMakeLists.txt:
set(CMAKE_BUILD_TYPE Debug)

Then rerun cmake and rebuild.  That should at least allow you to run it under gdb.  Once we know where the crash is we can debug.  It may be in OpenALPR code or it could be elsewhere.

**Comment by edwardscdr** [comment]
Rebuilt ok now but still same error:

gdb ./alprd
GNU gdb (Raspbian 7.7.1+dfsg-5) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "arm-linux-gnueabihf".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./alprd...done.
(gdb) set args -f
(gdb) run
Starting program: /home/pi/builds/openalpr/src/alprd -f
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".
Cannot access memory at address 0x0

Program received signal SIGILL, Illegal instruction.
0x74249d08 in ?? () from /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0
(gdb)
